test/helpers.js

```diff
@@ -1,29 +1,16 @@
 import test from 'ava';
 import got from '../source';
-import {createServer} from './helpers/server';
+import {withServer} from './helpers/with-server';
 
-let s;
-
-test.before('setup', async () => {
-	s = await createServer();
-
-	s.on('/', (request, response) => {
+test('promise mode', withServer, async (t, s) => {
```

Comment:
If you continue this, use `server` instead of `s`. It's more readable.

Reply:
Going to change it in all tests.